cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/KafkaConnectService.java

```diff
@@ -0,0 +1,176 @@
+            + "/config";
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(
+            suffixUrl, clusterConnectorRequest.getProtocol(), KafkaClustersType.KAFKA_CONNECT);
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Content-Type", "application/json");
+
+    HttpEntity<String> request =
+        new HttpEntity<>(clusterConnectorRequest.getConnectorConfig(), headers);
+
+    try {
+      reqDetails.getRight().put(reqDetails.getLeft(), request, String.class);
+    } catch (RestClientException e) {
+      log.error("Error in updating connector " + e.toString());
+      result.put("result", ApiResultStatus.ERROR.value);
+      result.put("errorText", e.toString().replaceAll("\"", ""));
+      return result;
+    }
+    result.put("result", ApiResultStatus.SUCCESS.value);
+
+    return result;
+  }
+
+  public ApiResponse postNewConnector(ClusterConnectorRequest clusterConnectorRequest)
+      throws Exception {
+    log.info("Into postNewConnector clusterConnectorRequest {} ", clusterConnectorRequest);
+
+    String suffixUrl = clusterConnectorRequest.getEnv() + "/connectors";
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(
+            suffixUrl, clusterConnectorRequest.getProtocol(), KafkaClustersType.KAFKA_CONNECT);
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Content-Type", "application/json");
+
+    HttpEntity<String> request =
+        new HttpEntity<>(clusterConnectorRequest.getConnectorConfig(), headers);
+    ResponseEntity<String> responseNew;
+    try {
+      responseNew =
+          reqDetails.getRight().postForEntity(reqDetails.getLeft(), request, String.class);
+    } catch (RestClientException e) {
+      log.error("Error in registering new connector " + e.toString());
+      throw new Exception(e.toString());
+    }
+    if (responseNew.getStatusCodeValue() == 201) {
+      return ApiResponse.builder().result(ApiResultStatus.SUCCESS.value).build();
+    } else {
+      return ApiResponse.builder().result(ApiResultStatus.FAILURE.value).build();
+    }
+  }
+
+  public List<String> getConnectors(String environmentVal, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into getConnectors {} {}", environmentVal, protocol);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String suffixUrl = environmentVal + "/connectors";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.KAFKA_CONNECT);
+
+      Map<String, String> params = new HashMap<>();
+      ResponseEntity<List> responseList =
+          reqDetails.getRight().getForEntity(reqDetails.getLeft(), List.class, params);
+
+      log.info("connectors list " + responseList);
+      return responseList.getBody();
+    } catch (Exception e) {
+      log.error("Error in getting connectors " + e.getMessage());
```

Comment:
```suggestion
      log.error("Error in getting connectors ", e);
```
it's better to log the whole stacktrace